### PR TITLE
Add BC linter

### PR DESCRIPTION
**Pull request by NicolasHug**
Add [BC linter](https://github.com/pytorch/test-infra/wiki/BC-Linter)

I just  copy pasted https://github.com/pytorch/pytorch/blob/main/.github/workflows/lint-bc.yml as is, hopefully it just works?

### CODE REVIEW

**Comment by izaitsevfb** [comment]
Yep, should just work.

**Review by pmeier — COMMENTED** [review]
I know the file is copy-pasted from core, but I don't see a strong reason why we couldn't put the job into https://github.com/pytorch/vision/blob/main/.github/workflows/lint.yml so we have all the linters in one place. We only need to guard this step to only run on PRs, since we need for parametrizing the action.

File: .github/workflows/lint-bc.yml
Comment (NicolasHug) [reply .github/workflows/lint-bc.yml:20]: > I don't think we need this

The whole block or just the `cancel-in-progress` part?

File: .github/workflows/lint-bc.yml
Comment (NicolasHug) [inline .github/workflows/lint-bc.yml:20]:
```suggestion
```

Comment (NicolasHug) [inline .github/workflows/lint-bc.yml:20]:
```suggestion
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}

```

File: .github/workflows/lint-bc.yml
Comment (pmeier) [reply .github/workflows/lint-bc.yml:20]: The whole block.

**Comment by pmeier** [comment]
Runs as expected: https://github.com/pytorch/vision/actions/runs/4935586144/jobs/8822079047?pr=7569

File: .github/workflows/lint.yml
Comment (pmeier) [inline .github/workflows/lint.yml:105]: Should we maybe switch to fixed version here, i.e.

```suggestion
        uses: pytorch/test-infra/.github/actions/bc-lint@v20230509-105021
```

or

```suggestion
        uses: pytorch/test-infra/.github/actions/bc-lint@d8944a4b2a96b70f66c3c56d97b93c2c6bc9936a
```

We are pinning the linter versions for everything else as well to avoid issues coming from upstream.

**Review by pmeier — APPROVED** [review]
Somewhat self-approving now. Happy to wait for someoneelses approval. Don't forget to revert e1ce69045fa187a94af5cb1a219bb63233c7c1b1 before merge.

File: .github/workflows/lint.yml
Comment (NicolasHug) [reply .github/workflows/lint.yml:105]: IMO keep main so we can benefits from improvements automatically. If it becomes noisy because of failures we can revisit

File: .github/workflows/lint-bc.yml
Comment (izaitsevfb) [reply .github/workflows/lint-bc.yml]: The reason for that is to allow suppression by label.

```
- opened
- synchronize
- reopened
```
Are the default types for `on / pull_request`, while `labeled` / `unlabeled` retrigger the job when `suppress-bc-linter` label is added.

**Pull request merged**
